File: clients/python/model/vec2_float.py
class Vec2Float:
    def __init__(self, x, y):
        self.x = x
        self.y = y
    @staticmethod
    def read_from(stream):
        x = stream.read_float()
        y = stream.read_float()
        return Vec2Float(x, y)
    def write_to(self, stream):
        stream.write_float(self.x)
        stream.write_float(self.y)
    def __repr__(self):
        return "Vec2Float(" + \
            repr(self.x) + "," + \
            repr(self.y) + \
            ")"

File: clients/ruby/model/vec2_float.rb
class Vec2Float
    attr_accessor :x
    attr_accessor :y
    def initialize(x, y)
        @x = x
        @y = y
    end
    def self.read_from(stream)
        x = stream.read_float()
        y = stream.read_float()
        Vec2Float.new(x, y)
    end
    def write_to(stream)
        stream.write_float(@x)
        stream.write_float(@y)
    end
end
